lib/libc/pthread/pthread_mutexattr_setprotocol.c

```diff
@@ -80,25 +80,26 @@
  *
  ****************************************************************************/
 
-int pthread_mutexattr_setprotocol(FAR pthread_mutexattr_t *attr,
-				  int protocol)
+int pthread_mutexattr_setprotocol(FAR pthread_mutexattr_t *attr, int protocol)
 {
 	svdbg("attr=0x%p protocol=%d\n", attr, protocol);
-	DEBUGASSERT(attr != NULL);
```

Comment:
DEBUGASSERT is only working when CONFIG_DEBUG is enabled.
Most of the case, we disable the debug option, so I think it is needed